### ISSUE TITLE
[DEV] clang version support

### ISSUE THREAD

**Issue by FrancescoBorzi** [azerothcore/forum#48]
Currently, in the [AC travis build](http://www.azerothcore.org/wiki/travis-build) we run 2 pipelines:

- id 1 is clang **3.8**
- id 2 is clang **7**

also we currently target **C++14**. There have been discussions about dropping the **3.8** version. Not worrying of such an old version of the compiler would bring several benefits.

We should decide what would be then the minimum version of clang that we want to support.

In my opinion we should try to make it at least version 5 so we can target C++17:

![image](https://user-images.githubusercontent.com/75517/63636122-73e99900-c66b-11e9-843b-8bdf49943a28.png)

(more info [here](https://clang.llvm.org/cxx_status.html))

In [ubuntu packages](https://packages.ubuntu.com/search?keywords=clang), the latest version that you can install in 16.04 (without adding extra repositories) looks like to be clang 6, so I think it is safe enough to support at least version 6 (that is also the default version in ubuntu 18.04)

**Comment by Viste** [comment]
we need C++17 for some good stuff so

**Comment by FrancescoBorzi** [comment]
I've opened a PR here where I change the min version of clang from 3.8 to 6.0:

https://github.com/azerothcore/azerothcore-wotlk/pull/2232

if you agree with this, just approve that PR, otherwise let's discuss it here

**Comment by Yehonal** [comment]
with 6.0 we cannot support ubuntu 16.04 anymore, it's good for me but we've to know it

**Comment by FrancescoBorzi** [comment]
@Yehonal why not? as I said:

> In ubuntu packages, the latest version that you can install in 16.04 (without adding extra repositories) looks like to be clang 6

See:

- https://packages.ubuntu.com/search?keywords=clang
- https://packages.ubuntu.com/xenial-updates/clang-6.0

**Comment by Stoabrogga** [comment]
I use clang 7 on Ubuntu 16.04 via an additional repository, but clang 6 is available in the default repositories:
```
# apt-cache search clang-6.0
clang-6.0 - C, C++ and Objective-C compiler
clang-6.0-doc - C, C++ and Objective-C compiler - Documentation
clang-6.0-examples - Clang examples
libclang-6.0-dev - clang library - Development package
python-clang-6.0 - Clang Python Bindings
```

**Comment by Yehonal** [comment]
Ok my mistake, i've done a research yesteday and maybe i missed the 5.0 and 6.0 dunno why ^_^'

**Comment by FrancescoBorzi** [comment]
https://github.com/azerothcore/wiki/pull/117/files

**Comment by BarbzYHOOL** [comment]
what is the available clang on deb 9 and ub 16.04?? because if only 3.8 can work, it's needed

**Comment by Viste** [comment]
> what is the available clang on deb 9 and ub 16.04?? because if only 3.8 can work, it's needed

by default 3.8 but you can easy update to 6. 
But 3.8 clang more bugged  then 6 so use 3.8 is bad idea :confused:
